src/iop/colorzones.c

```diff
@@ -372,8 +372,8 @@ void process_v3(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, con
   }
 }
 
-void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const void *const ivoid,
-             void *const ovoid, const dt_iop_roi_t *const roi_in, const dt_iop_roi_t *const roi_out)
+void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const void *const restrict ivoid,
+             void *const restrict ovoid, const dt_iop_roi_t *const roi_in, const dt_iop_roi_t *const roi_out)
```

Comment:
I don't think restrict belongs here.

Reply:
restrict is just a way to prevent the compiler from detecting false aliasing and may enable further optimizations. It should work this way (already tested in other IOPs).

Reply:
Not sure if process() should be different from its definition, but in any case it should go in a separate PR, or at least on a separate commit.

Reply:
interesting indeed that the "restrict" keyword doesn't change the type of the function pointer. since it's a good thing to have here (some vectorisation will insist to have it or else fail) i agree we should change the function definitions around here https://github.com/darktable-org/darktable/blob/master/src/develop/imageop.h#L256

Reply:
ok I removed the restrict keyword, let's do that elswhere